### PR TITLE
Admin Generator (Future): Add settings to restrict user-actions in grid generator

### PR DESCRIPTION
**Pull request by jamesricky**
- Individual settings `add`,  `edit`, `delete`, `copyPaste` to disable those individual features, they are enabled by default. 
- Additional `readOnly` setting to disable all those features at once.

### CODE REVIEW

**Comment by nsams** [comment]
- :+1: `disableCopyPaste`
    - I'm not sure about the naming, why not `copyPaste=false` with defaulting to the current implementation

- `readOnly`:
    - a readOnly detail page might be required even if the data is read only
    - maybe could be split into delete/create/edit

**Comment by johnnyomair** [comment]
> I'm not sure about the naming, why not copyPaste=false with defaulting to the current implementation

I'd also use `copyPaste` without the `disable` prefix. The prefix is handy for component props since it defaults to true (e.g. `<Button disableHover />`, but in the grid config it adds unnecessary complexity: `disableCopyPaste: true` vs `copyPaste: false`

**Comment by johnnyomair** [comment]
It's a good idea to combine readonly and copyPaste. I'm not sure if `interaction` is the best name for this, but I can't think of something better (`readOnly` would be more common, but it's not that fitting).

Instead of having an object we could also use multiple options and do something like this:

```ts
const readOnly = {
  edit: false,
  delete: false, 
  add: false,
  copyPaste: false
} 

const config = {
  ...readOnly
}
```

What do you think?

**Comment by jamesricky** [comment]
> I'm not sure if `interaction` is the best name for this, but I can't think of something better (`readOnly` would be more common, but it's not that fitting).

I agree that `interaction` is a bad name but I also don't like `readOnly` it's not really fitting... also can't think of anything better though 😅 

> Instead of having an object we could also use multiple options

I prefer having one value that can be set to either a boolean to disable everything with just one setting/line, or an object to disable only specific things. 

Also having a setting named `edit` sounds to me like this is the edit-function, not a boolean to enable/disable editing of the rows.

**Comment by nsams** [comment]
easiest to use would be to support both:
```
const config1: GridConfig = {
  foo: "bar",
  readOnly: true,
}
const config2: GridConfig = {
  foo: "bar",
  delete: false,
}
```
that can even be [typed correctly ](https://www.typescriptlang.org/play?#code/C4TwDgpgBA4gTgSwCYGED2A7AZgg5lAXigG8os00AuKAZ2EQ1wG4oBfKAMigApiAoKFDgQAhkgDyGADYgA-NQBGFKaIxM+7AD4kBQ0ROkhqAVwxIIODBCTrB1hMHlQlaFSLW7zK4BCcu3HoJiSH7KqrZQAMZoYCAACiJ0voph7uqsAJTqfHzRGHRRmDi4AIzU8Mjo2HiEOoLkVFAARAoicE0ANLrCYpIy1PTGEF2suZgFecUATOWIqEU1RPz1FNQtbZ2eEN4Q1FgiUjTDGjl5Ewu4AMyzlRe1y2Srza3tXYI9Bv1Qg8eCAPR-KAAAWANAAtBAAB6QSLACFwOBoOBbHZ7A5HEZ8IA)

**Comment by jamesricky** [comment]
> > Also having a setting named edit sounds to me like this is the edit-function, not a boolean to enable/disable editing of the rows.
> 
> I don't see a difference between disabling an "edit-function" and "disable editing of the rows"

What I meant was that a config-setting named "edit" sounds like I should pass in a function that handles the edit-functionality, not like a setting to enable/disabled editing of the row in the UI. That's why I wanted to nest it in something to make it more obvious what this setting does... but maybe that's just me 🤷🏼‍♂️

File: demo/admin/src/products/future/generated/ProductsGrid.tsx [hunk 1]
```diff
@@ -152,6 +152,7 @@ export function ProductsGrid(): React.ReactElement {
                         <IconButton component={StackLink} pageName="edit" payload={params.row.id}>
                             <Edit color="primary" />
                         </IconButton>
+
```
Comment (johnnyomair) [inline demo/admin/src/products/future/generated/ProductsGrid.tsx:155]: Could we remove the empty line here? 😅

Comment (jamesricky) [reply demo/admin/src/products/future/generated/ProductsGrid.tsx]: Makes the generator code look a bit weird but yes 😄

**Review by johnnyomair — DISMISSED** [review]

**Review by nsams — APPROVED** [review]

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**